crates/cairo-profiler/src/profile_builder.rs

```diff
@@ -267,3 +273,26 @@ pub fn build_profile(samples: &[Sample]) -> pprof::Profile {
         default_sample_type: 0,
     }
 }
+
+pub fn save_profile(target_path: &Utf8PathBuf, profile: &pprof::Profile) -> Result<()> {
```

Comment:
I believe this function shouldn't be in this module as it is specifically designed for converting profiler's internal representation to pprof representation. Adding saving to a file here mixes abstraction layers imo and makes the purpose of the module less clear